compiler/rustc_hir_analysis/src/check/check.rs

```diff
@@ -722,7 +722,9 @@ pub(super) fn check_specialization_validity<'tcx>(
     let result = opt_result.unwrap_or(Ok(()));
 
     if let Err(parent_impl) = result {
-        report_forbidden_specialization(tcx, impl_item, parent_impl);
+        if !tcx.is_impl_trait_in_trait(impl_item) {
```

Comment:
please delay a bug here

Reply:
Have delayed a bug, but please would rather if you check what I did what you meant and if the english wording is fine :).

Reply:
It's just a bug so it's fine.